app/components/header/index.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import Header from './index'
+
+jest.mock('next/navigation', () => ({
+  useRouter: jest.fn(),
+}))
+
+describe('Header component: test cases', () => {
+  test('header components element should be rendered', () => {
+    render(<Header />)
+    const buttonEl = screen.getByRole('button')
+    expect(buttonEl).toBeInTheDocument()
+  })
+
+  test('button text should be new todo', () => {
+    render(<Header />)
+    const buttonEl = screen.getByRole('button')
+    expect(buttonEl).toHaveTextContent('+ New Todo')
+  })
+
+  test.skip('new todo button click should re-direct to /new-item route', async () => {
+    render(<Header />)
+    const buttonEl = screen.getByRole('button')
+    fireEvent.click(buttonEl)
+    await waitFor(() => {
+      expect(global.window.location.href).toContain('/new-item')
```

Comment:
Since we are mocking the router, I don't think this would work. 
We can probably check if router is invoked with proper path.

Reply:
Updated the test case with invoked router path.